### PR TITLE
CTCP-2748: Try to explicitly send headers and prevent the HeaderCarrier from stripping required fields.

### PR DESCRIPTION
**Pull request by drnaylor**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnector.scala [hunk 1]
```diff
@@ -41,18 +48,30 @@ trait EISConnector {
...
+
   override def post(body: Source[ByteString, _])(implicit hc: HeaderCarrier): Future[Either[RoutingError, Unit]] =
     httpClientV2
       .post(url"${eisInstanceConfig.url}")
       .setHeader(
-        hc.headers(Seq("X-Conversation-Id", "X-Correlation-Id", "Content-Type", "Accept", "Date")): _*
+        hc.headers(Seq("X-Conversation-Id", "X-Correlation-Id")): _*
+      )
+      .setHeader(
+        HeaderNames.AUTHORIZATION -> hc.authorization.get.value,
```
Comment (sonisn1988) [inline app/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnector.scala:71]: Possibility of getting exception at this point if authorisation is not present in header carrier?

File: it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala [hunk 2]
```diff
@@ -72,14 +74,16 @@ class EISConnectorSpec
     uriStub
   )
 
+  val clock: Clock = Clock.fixed(LocalDateTime.of(2023, 5, 4, 12, 36, 8, 0).toInstant(ZoneOffset.UTC), ZoneOffset.UTC)
```
Comment (sonisn1988) [inline it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala:77]: Its better to comments something about these number 2023, 5, 4, 12, 36, 8, 0. to make more sense to anyone looking on this.

Comment (drnaylor) [reply it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala:77]: It's quite clear what they are from the signature.

File: app/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnector.scala (hunk 1 above)
Comment (drnaylor) [reply app/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnector.scala:71]: If it's not there we have bigger problems - it must always be there.

**Review by sonisn1988 — APPROVED** [review]

**Pull request merged**